app/Services/EntityService.php

```diff
@@ -452,6 +452,11 @@ protected function moveType(Entity $entity, string $target, MiscModel $misc = nu
             }
         }
 
+        //Remove tags if converting to tag, since tags can have tags.
+        if ($new->entityTypeId() === config('entities.ids.tag')) {
+            $old->entity->tags()->detach();
```

Comment:
Instead of calling $old->entity, which creates a new db request for just this, use `$entity` directly instead